traits/has_traits.py

```diff
@@ -2047,7 +2028,47 @@ def class_trait_view_elements(cls):
         The returned object can be used to access all the view elements
         associated with the class.
         """
-        return cls.__dict__[ViewTraits]
+        view_elements = cls.__dict__[ViewTraits]
+        if isinstance(view_elements, dict):
+            view_elements = cls._init_trait_view_elements()
+        return view_elements
+
+    @classmethod
+    def _init_trait_view_elements(cls):
+        """ Lazily Initialize the ViewElements object from a dictionary. """
+        from traitsui.view_elements import ViewElements
+
+        hastraits_bases = [
+            base for base in cls.__bases__
+            if base.__dict__.get(ClassTraits) is not None
```

Comment:
would be simpler as `if ClassTraits in base.__dict__`

Reply:
What about `hasattr(base, ClassTraits)`?  I assume that is safely equivalent.

Reply:
Assuming no excess of magic, yes, should be safely equivalent. (In *theory*, `hasattr` does the equivalent of a `try` / `except` on `getattr`, which could be doing weird and wonderful things in the presence of `__getattribute__` and friends. But we're not overriding `__getattribute__` on the metaclass; only on the classes.)

Reply:
From offline discussion - it isn't equivalent in general because we don't want to see superclass attributes - although in practice I think it would be fine; `issubclass(base, HasTraits)`.